src/unix/kqueue.c

```diff
@@ -488,11 +488,15 @@ static void uv__fs_event(uv_loop_t* loop, uv__io_t* w, unsigned int fflags) {
    * the struct's kf_structsize must be initialised beforehand
    * whether with the KINFO_FILE_SIZE constant or this way.
    */
-  struct kinfo_file kf;
-  kf.kf_structsize = sizeof(kf);
+  struct stat statbuf;
 
-  if (fcntl(handle->event_watcher.fd, F_KINFO, &kf) == 0)
-    path = uv__basename_r(kf.kf_path);
+  if (handle->event_watcher.fd != -1 &&
+     (!uv__fstat(handle->event_watcher.fd, &statbuf) && !(statbuf.st_mode & S_IFDIR))) {
+	  struct kinfo_file kf;
+	  kf.kf_structsize = KINFO_FILE_SIZE;
+	  if (fcntl(handle->event_watcher.fd, F_KINFO, &kf) == 0)
+		  path = uv__basename_r(kf.kf_path);
```

Comment:
Style: indentation is off, should be two spaces.

Substance: can you explain how calling fstat makes a difference? If F_KINFO doesn't handle directories, then I assume it returns something like EISDIR, resulting in `path` not being assigned. Calling fstat first wouldn't change that, ISTM.

@rhelmot maybe you can chime in / try out this patch?

Reply:
This patch causes tests to pass on my freebsd14 machine!

Reply:
@rhelmot thanks for testing

@devnexen can you comment on this?

>  can you explain how calling fstat makes a difference?

Reply:
No fcntl in this case does not return `EISDIR`, to be fair F_KINFO is pretty still a new feature and not wildly used yet.